vfs/fs_utils.h

```diff
@@ -48,10 +48,22 @@ struct FsObjectInfo {
     std::map<std::string, Json::Value> objectMetadata;  ///< Metadata from obj
 
     operator bool () const { return exists; }
+
+    // make dynamic_pointer_cast work otherwise not detected as polymorphism
+    // struct
+    virtual ~FsObjectInfo() {}
```

Comment:
Can't do this; the class is designed to be value copyable.  This will lead to strange and intermittent bugs due to object slicing.